### ISSUE TITLE
General speedup revisited

### ISSUE THREAD

**Issue by connie** [ReactionMechanismGenerator/RMG-Py#471]
See https://github.com/ReactionMechanismGenerator/RMG-Py/issues/78 for the older version of this issue.

I've noticed that on my computer, the minimal example running on 534cf3af829  takes 42 seconds vs the latest master at 21b83e3a takes 47 seconds, making it 5 seconds slower.  I imagine this slowdown will affect larger models as well.  @KEHANG presented at group meeting some changes which can help speed things up.  I think there are also portions of code that could be cythonized or re-optimized.  Please use this issue to brainstorm ideas or provide new profiling data.

**Comment by nickvandewiele** [comment]
#488 should make simulations with many edge reactions faster.

**Comment by nickvandewiele** [comment]
![cpu_eg1](https://cloud.githubusercontent.com/assets/242147/11077449/368cda02-87ce-11e5-82bf-2e4aafd57e11.png)

The figures shows the CPU times of `make eg1` of the last 300 Travis CI builds triggered by `ReactionMechanismGenerator/RMG-Py`, i.e. pull requests into master and new branches.

**Comment by nickvandewiele** [comment]
![memory_used_eg1](https://cloud.githubusercontent.com/assets/242147/11077595/eaa84e22-87ce-11e5-9dfc-153a9a579089.png)

Same for RAM used by `make eg1`

**Comment by rwest** [comment]
@walsh-p and @sherman-sc ran this job for about 7 hours before it crashed (an ODE solver problem). It got to 126 core species.
Here is the [RMG.profile.pdf](https://github.com/ReactionMechanismGenerator/RMG-Py/files/32575/RMG.profile.pdf)

Two remarkably large chunks:
 * chemkin:1653:markDuplicateReaction 23.16% (23.17%) 135941×
 * method 'update' of 'rmgpy.molecule.molecule.Molecule' objects> 27.61% (27.62%) 7599900×

Can we speed them up or call them less frequently?
Looks like we applied reaction recipes 2.5M times, so ran `updateConnectivityValues` 5M times (on average 2 new products per reaction?), but *also* ran `update` 7.5M times?
